python/lsst/ip/isr/isrFunctions.py

```diff
@@ -502,197 +502,19 @@ def overscanCorrection(ampMaskedImage, overscanImage, fitType='MEDIAN', order=1,
-            axes.plot(indices, fitBiasArr, 'r-')
-            plot.xlabel("centered/scaled position along overscan region")
-            plot.ylabel("pixel value/fit value")
-            figure.show()
-            prompt = "Press Enter or c to continue [chp]... "
-            while True:
-                ans = input(prompt).lower()
-                if ans in ("", "c",):
-                    break
-                if ans in ("p",):
-                    import pdb
-                    pdb.set_trace()
-                elif ans in ("h", ):
-                    print("h[elp] c[ontinue] p[db]")
-            plot.close()
-
-        offImage = ampImage.Factory(ampImage.getDimensions())
-        offArray = offImage.getArray()
-        overscanFit = afwImage.ImageF(overscanImage.getDimensions())
-        overscanArray = overscanFit.getArray()
-        if shortInd == 1:
-            offArray[:, :] = fitBiasArr[:, numpy.newaxis]
-            overscanArray[:, :] = fitBiasArr[:, numpy.newaxis]
-        else:
-            offArray[:, :] = fitBiasArr[numpy.newaxis, :]
-            overscanArray[:, :] = fitBiasArr[numpy.newaxis, :]
-
-        # We don't trust any extrapolation: mask those pixels as SUSPECT
-        # This will occur when the top and or bottom edges of the overscan
-        # contain saturated values. The values will be extrapolated from
-        # the surrounding pixels, but we cannot entirely trust the value of
-        # the extrapolation, and will mark the image mask plane to flag the
-        # image as such.
-        mask = ampMaskedImage.getMask()
-        maskArray = mask.getArray() if shortInd == 1 else mask.getArray().transpose()
-        suspect = mask.getPlaneBitMask("SUSPECT")
-        try:
-            if collapsed.mask == numpy.ma.nomask:
-                # There is no mask, so the whole array is fine
-                pass
-        except ValueError:      # If collapsed.mask is an array the test fails [needs .all()]
-            for low in range(num):
-                if not collapsed.mask[low]:
-                    break
-            if low > 0:
-                maskArray[:low, :] |= suspect
-            for high in range(1, num):
-                if not collapsed.mask[-high]:
-                    break
-            if high > 1:
-                maskArray[-high:, :] |= suspect
 
-    else:
-        raise pexExcept.Exception('%s : %s an invalid overscan type' % ("overscanCorrection", fitType))
-    ampImage -= offImage
-    overscanImage -= overscanFit
-    return Struct(imageFit=offImage, overscanFit=overscanFit, overscanImage=overscanImage)
+    config = OverscanCorrectionTaskConfig()
+    if fitType:
+        config.fitType = fitType
+    if order:
+        config.order = order
+    if collapseRej:
+        config.numSigmaClip = collapseRej
+    if overscanIsInt:
+        config.overscanIsInt = True
+
+    overscanTask = OverscanCorrectionTask(config=config)
+    return overscanTask.run(ampImage, overscanImage)
```

Comment:
You should deprecate this function, and modify `IsrTask` to call the `OverscanCorrectionTask` directly. Otherwise, there's not really any advantage to having the overscan correction in a `Task`.

Reply:
The newest commit does this, although it still calls it from an IsrTask.overscanCorrection call.  The overscan chopping is still in IsrTask, allowing OverscanCorrectionTask to concentrate on the image-level math.

This has necessitated config changes on obs_ packages to point to the updated overscan config item.